### DISCUSSION TITLE
Change $subscribe to a different name to not conflict with rx implementations

### DISCUSSION THREAD

**Discussion by ScottAwesome** [vuejs/pinia#1003] [Ideas]
This is really an ergonomics idea.

If you use any of the major observables implementation, like [rxjs](https://rxjs.dev/) they all  use `subscribe` and `unsubscribe` to mean something with observables.

As an ergonomic standpoint, having both `$subscribe` and `subscribe` (and in some cases, if you follow the RX nomeclature, observables themselves may be named `observed$` so you end up with something like `observed$.subscribe`), things become confusing to quickly grasp what you are or are not subscribing to in a code base.

The though here being so we don't have these naming clashes, is to rename `$subscribe` to `$observe` there by removing the nomenclature confusion.

I realize this is likely a small lift, however it could be done via deprecation and just aliasing `$subscribe` to `$observe`

Alternative naming of course, is welcome!

My hope is this makes it just a little more ergonomic to use `rxjs` and others with pinia

**Comment by posva** [comment]
Sorry for the late reply. I like `$subscribe()`, I think it's intuitive for people, conflicting with other libraries is also part of what makes it intuitive in the first place
